docs/source/basic_tutorials/launch.mdx

```diff
@@ -0,0 +1,148 @@
+    accelerator.backward(loss)
+    optimizer.step()
+    scheduler.step()
+```
+
+But how do you run this code and have it utilize the special hardware available to it?
+
+First you should rewrite the above code into a function, and make it callable as a script. For example:
+
+```diff
+  from accelerate import Accelerator
+  
++ def main():
+
+    accelerator = Accelerator()
+
+    model, optimizer, training_dataloader, scheduler = accelerator.prepare(
+        model, optimizer, training_dataloader, scheduler
+    )
+
+    for batch in training_dataloader:
+        optimizer.zero_grad()
+        inputs, targets = batch
+        outputs = model(inputs)
+        loss = loss_function(outputs, targets)
+        accelerator.backward(loss)
+        optimizer.step()
+        scheduler.step()
+
++ if __name__ == "__main__":
++   main()
+```
+
+Next you need to launch it with `accelerate launch`. 
+
+## Using accelerate launch
+
+🤗 Accelerate has a special CLI command to help you launch your code in your system through `accelerate launch`.
+This command wraps around all of the different commands needed to launch your script on various platforms, without you
+having to remember what each of them are.
+
+<Tip>
+
+If you are familiar with launching scripts in PyTorch yourself such as with `torchrun`, you can still do this. It is not required to use `accelerate launch`
+
+</Tip>
+
+You can launch your script quickly by using:
+```bash
+accelerate launch {script_name.py} --arg1 --arg2 ...
+```
+
+Just put `accelerate launch` at the start of your command, and pass in additional arguments and parameters to your script afterwards like normal!
+
+Since this runs the various torch spawn methods, all of the expected environment variables can be modified here as well.
+For example, here is how to use `accelerate launch` with a single GPU:
+```bash
+CUDA_VISIBLE_DEVICES="0" accelerate launch {script_name.py} --arg1 --arg2 ...
+```
+
+You can also use `accelerate launch` without performing `accelerate config` first, but you may need to manually pass in the right configuration parameters.
+In this case, 🤗 Accelerate will make some hyperparameter decisions for you, such as if a GPU is available, it will use all of them by default, and no mixed precision will be used. 
+Here is how you would use all GPUs and train on no mixed precision:
+```bash
+accelerate launch --multi_gpu {script_name.py} --arg1 --arg2 ...
+```
+To get more specific you should pass in the needed parameters yourself. For instance, here is how you 
+would also launch that same script on two GPUs using mixed precision while avoiding all of the warnings: 
+```bash
+accelerate launch --multi_gpu --mixed_precision=fp16 --num_processes=2 {script_name.py} --arg1 --arg2 ...
+```
+
+For a complete list of parameters you can pass in, run:
+```bash
+accelerate launch -h
+```
+
+<Tip>
+
+Even if you are not using 🤗 Accelerate in your code, you can still use the launcher for starting your scripts!
```

Comment:
what's the benefit of that? what happens if you start a random script with accelerate?

Reply:
It will remove the boilerplate needed when calling `torchrun` XYZ. I can expand this further to point it out, by translating the earlier bash with accelerate launch to show the equivalence.